pkg/execution/executor/executor.go

```diff
@@ -937,7 +946,18 @@ func (e *executor) Cancel(ctx context.Context, runID ulid.ULID, r execution.Canc
 
 	// TODO: Load all pauses for the function and remove, once we index pauses.
 
-	s, _ := e.sm.Load(ctx, runID)
+	s, err := e.sm.Load(ctx, runID)
+	if err != nil {
```

Comment:
We call `e.sm.Metadata` at the beginning of this function, so we may as well call `e.sm.Load` at the top and get Metadata out of the loaded state.  This reduces the number of calls to the state store.

Reply:
Yeah, also weird that we load state after canceling.